controller_manager/src/controller_manager.cpp

```diff
@@ -188,6 +202,27 @@ ControllerManager::ControllerManager(
   init_services();
 }
 
+void ControllerManager::robot_description_callback(const std_msgs::msg::String & robot_description)
+{
+  RCLCPP_INFO(get_logger(), "Received robot description file.");
+  RCLCPP_DEBUG(
+    get_logger(), "'Content of robot description file: %s", robot_description.data.c_str());
+  // TODO(Manuel) errors should probably be caught since we don't want controller_manager node
+  // to die if a non valid urdf is passed. However, should maybe be fine tuned.
+  try
+  {
+    init_resource_manager(robot_description.data.c_str());
```

Comment:
As you suggested, this line should only be executed once. You can implement it maybe via a `ResourceManager.is_initialized()` function

Reply:
Done could someone please check if this is how it should be